### PR TITLE
cluster-capi-kind package

### PR DESCRIPTION
**Pull request by henderiw**
package to deploy a kind cluster using capi

### CODE REVIEW

File: cluster-capi-kind/cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        audit: restricted
+        enabled: true
+        enforce: baseline
+        warn: restricted
+    version: v1.26.3
+    workers:
+      machineDeployments:
+      - class: default-worker
+        name: md-0
+        replicas: 3
```
Comment (johnbelamaric) [inline cluster-capi-kind/cluster.yaml]: missing newline

**Review by johnbelamaric — COMMENTED** [review]
Some missing newlines but otherwise LGTM.

It will be interesting to experiment with how we can do mutations to things like pod and service cidrs via the PackageVariants that consume this package.

File: cluster-capi-kind/Kptfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: kpt.dev/v1
+kind: Kptfile
+metadata:
+  name: capi-kind-cluster
+  annotations:
+    config.kubernetes.io/local-config: "true"
+info:
+  description: sample description
```
Comment (electrocucaracha) [inline cluster-capi-kind/Kptfile]:
```suggestion
  description: This provides a blueprint manifest to deploy a kind cluster using cluster api
```

File: cluster-capi-kind/cluster.yaml (hunk 1 above)
Comment (henderiw) [reply cluster-capi-kind/cluster.yaml]: fixed

File: cluster-capi-kind/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# cluster
+
+## Description
+
+This package provides a blueprint manifest to deploy a kind cluster using cluster api
```
Comment (electrocucaracha) [inline cluster-capi-kind/README.md:5]: It should be nice to include some defaults values in the documentation like kubernetes version, pod/service cirdrs, number of workers, etc.

File: cluster-capi-kind/cluster.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    - name: imageRepository
+      value: ""
+    - name: etcdImageTag
+      value: ""
+    - name: coreDNSImageTag
+      value: ""
+    - name: podSecurityStandard
+      value:
+        audit: restricted
+        enabled: true
```
Comment (electrocucaracha) [inline cluster-capi-kind/cluster.yaml:30]: I faced some issues during the free5GC deployment with `podSecurityStandard` feature. I'm in favor to have it enabled by default, but I just want to mention it.

File: cluster-capi-kind/README.md (hunk 3 above)
Comment (electrocucaracha) [inline cluster-capi-kind/README.md:5]: Another thing to document are the host dependencies like, `docker` and the `cni plugins` folder location in the host.

**Comment by gvbalaji** [comment]
Other than Victor's comments, 
/approve

**Comment by johnbelamaric** [comment]
Same here.

/approve

File: cluster-capi-kind/Kptfile (hunk 2 above)
Comment (henderiw) [reply cluster-capi-kind/Kptfile]: fixed

File: cluster-capi-kind/README.md (hunk 3 above)
Comment (henderiw) [reply cluster-capi-kind/README.md:5]: I added this:

The package contains some defaults but can be changed through the kpt pipeline
- pod cidrBlocks: 192.168.0.0/16
- service cidrBlocks: 10.128.0.0/12
- service domain: cluster.local
- kubernetes version: v1.26.3
- workers: 3

The package assumes that we mount through /opt/cni/bin the cni(s)

File: cluster-capi-kind/cluster.yaml (hunk 4 above)
Comment (henderiw) [reply cluster-capi-kind/cluster.yaml:30]: ok I took this from the templates they generate

File: cluster-capi-kind/README.md (hunk 3 above)
Comment (henderiw) [reply cluster-capi-kind/README.md:5]: The package has some host dependencies:
- /opt/cni/bin for accessing the cni(s)
- /var/run/docker.sock for the docker socket

added to the readme

File: cluster-capi-kind/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+The package contains some defaults but can be changed through the kpt pipeline
+- pod cidrBlocks: 192.168.0.0/16
+- service cidrBlocks: 10.128.0.0/12
+- service domain: cluster.local
+- kubernetes version: v1.26.3
+- workers: 3
+
+The package has some host dependencies:
+- /opt/cni/bin for accessing the cni(s)
+- /var/run/docker.sock for the docker socket
```
Comment (vjayaramrh) [inline cluster-capi-kind/README.md:16]: May need podman support as well, we can probably create a separate issue to track podman support

File: cluster-capi-kind/docker_machine_template_control_plane.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: DockerMachineTemplate
+metadata:
+  name: nephio-control-plane
+  namespace: default
+spec:
+  template:
+    spec:
+      extraMounts:
+      - containerPath: /var/run/docker.sock
```
Comment (vjayaramrh) [inline cluster-capi-kind/docker_machine_template_control_plane.yaml:10]: Wondering if mutator pipeline can be used to replace this with podman related values if podman support is desired

Comment (johnbelamaric) [reply cluster-capi-kind/docker_machine_template_control_plane.yaml:10]: Yeah, that will be an interesting test case for how we might do that!

**Comment by johnbelamaric** [comment]
/lgtm

Let's address anything more in future PRs.

File: cluster-capi-kind/cluster_class.yaml
Comment (johnbelamaric) [inline cluster-capi-kind/cluster_class.yaml:1]: Should this be a one-per-management-cluster resource, not one-per-workload-cluster?

**Pull request merged**